### PR TITLE
support multi-arch docker images

### PR DESCRIPTION
**Pull request by better0332**
### What problem does this PR solve?
Support multi-arch docker images

### Code changes

- [ ] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,8 +42,8 @@ ifeq ($(E2E),y)
 	docker build --tag "${DOCKER_REPO}/tidb-operator:${IMAGE_TAG}" -f images/tidb-operator/Dockerfile.e2e images/tidb-operator
 	docker build --tag "${DOCKER_REPO}/tidb-backup-manager:${IMAGE_TAG}" -f images/tidb-backup-manager/Dockerfile.e2e images/tidb-backup-manager
 else
-	docker build --tag "${DOCKER_REPO}/tidb-operator:${IMAGE_TAG}" images/tidb-operator
-	docker build --tag "${DOCKER_REPO}/tidb-backup-manager:${IMAGE_TAG}" images/tidb-backup-manager
+	docker build --tag "${DOCKER_REPO}/tidb-operator:${IMAGE_TAG}" --build-arg=TARGETARCH=$(GOARCH) images/tidb-operator
```
Comment (liubog2008) [inline Makefile:45]: `TARGETARCH` is auto set by `--platform`

File: Makefile [hunk 2]
```diff
@@ -42,8 +42,8 @@ ifeq ($(E2E),y)
...
+	docker build --tag "${DOCKER_REPO}/tidb-backup-manager:${IMAGE_TAG}" --build-arg=TARGETARCH=$(GOARCH) images/tidb-backup-manager
```
Comment (liubog2008) [inline Makefile:46]: When GOARCH is `amd64`, `TARGETARCH` should be `linux/amd64`

File: ci/release_tidb_operator_binary_and_image.groovy [hunk 3]
```diff
@@ -26,11 +26,17 @@ def call(BUILD_BRANCH, RELEASE_TAG, CREDENTIALS_ID, CHART_ITEMS) {
                     images.each {
                         stage("Build and push ${it} image") {
                             withDockerServer([uri: "${env.DOCKER_HOST}"]) {
-                                docker.build("pingcap/${it}:${RELEASE_TAG}", "images/${it}").push()
+                                sh """
+                                docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```
Comment (liubog2008) [inline ci/release_tidb_operator_binary_and_image.groovy:30]: Privileged container can be run in Jenkinsfile? It's a critical security issue...

File: images/tidb-backup-manager/Dockerfile [hunk 4]
```diff
@@ -1,30 +1,31 @@
-FROM pingcap/alpine-glibc:alpine-3.14
+FROM pingcap/alpine-glibc:alpine-3.14.3
```
Comment (KanShiori) [inline images/tidb-backup-manager/Dockerfile:1]: Can base image work in arm?

File: Makefile (hunk 1 above)
Comment (better0332) [reply Makefile:45]: `TARGETARCH` is auto set by buildx not build

File: ci/release_tidb_operator_binary_and_image.groovy (hunk 3 above)
Comment (better0332) [reply ci/release_tidb_operator_binary_and_image.groovy:30]: I also copied it from other articles. Let't me try

File: Makefile (hunk 2 above)
Comment (better0332) [reply Makefile:46]: no, it may `TARGETPLATFORM`, not `TARGETARCH`

File: images/tidb-backup-manager/Dockerfile (hunk 4 above)
Comment (better0332) [reply images/tidb-backup-manager/Dockerfile:1]: yes, you can try docker buildx imagetools inspect pingcap/alpine-glibc:alpine-3.14.3

File: ci/release_tidb_operator_binary_and_image.groovy (hunk 3 above)
Comment (KanShiori) [reply ci/release_tidb_operator_binary_and_image.groovy:30]: Maybe we need to use the image with a specific version and the version is configurable. Also, we use the image from our registry

**Review by liubog2008 — APPROVED** [review]
LGTM

**Pull request merged**